_posts/2015-09-30-september-newsletter.md

```diff
@@ -0,0 +1,274 @@
+greatly helped us focus on more urgent tasks that resulted in the first release.
+We planned many features not yet available in 0.1, but decided to postpone them
+for future milestones, which will be more frequent after this newsletter.
+
+This illustrates the path Neovim will take for now on: Instead of preparing
+big releases that take forever to happen, we'll focus on smaller, frequent and
+more stable releases.
+
+The 0.1 release is basically just a tag that users looking to compile Neovim in
+a version that has a minimum level of stability, but future releases may also
+contain precompiled binaries and even installers(when Windows is officially
+supported)
+
+### Bountysource salt campaign
+
+For those who don't know yet, Bountysource launched a new platform that allows
+open source projects to obtain sustainable crowdfunding. This platform is
+conveniently called ["salt"][history-of-salt], and Neovim was one of the first
+projects to use it.
+
+The [first campaign][first-campaign] was very
+successful and raised about $35,000, which allowed me to work full-time on
+Neovim for roughly 6 months. Being very enthusiastic about the project and
+unable to meet all goals in those months, I continued to dedicate a very
+significant portion of my time to Neovim, so much that it started hurting my
+personal and professional life. This continued until February when I saw that I
+simply couldn't continue with my old pace. Around that time that @rappo offered
+me to test the salt platform beta version and I saw it as a way to continue my
+work on Neovim.
+
+Like it's predecessor, [the salt campaign][salt-campaign] was very successful
+and allowed me to continue Neovim contributions(in a healthy way) for the past 5
+months, thank you!
+
+### Building Neovim from source
+
+Did we ever mention how easy it is to build and install Neovim from source?
+While it has a good number of dependencies, the build system automatically
+downloads and builds everything without cluttering your system. Here's a quick
+tutorial on how to install Neovim on a unix system that has git, cmake, gnu
+autotools and a recent version of GCC:
+
+```sh
+git clone https://github.com/neovim/neovim
+cd neovim
+git checkout 0.1   # optional step
+make CMAKE_EXTRA_FLAGS="-DCMAKE_INSTALL_PREFIX=$HOME/.neovim" install
+sudo ln -s $HOME/.neovim/bin/nvim /usr/local/bin/nvim
+```
+
+And uninstalling:
+
+```sh
+rm -rf $HOME/.neovim
+sudo rm /usr/local/bin/nvim
+```
+
+That's it, go try it yourself!
+
+## Development News
+
+### SHADA(SHAred DAta)
+
+@ZyX-I is working on a [major PR][shada-pr] that will completely replace the
+viminfo file for storing user data such as register contents, command history,
+variables, jump list and so on.
+
+A [known problem][shada-proposal] with viminfo is that two Vim instances running
+concurrently will override each other's data. SHADA is a new storage format
+created by @ZyX-I that will bring a number of enhancements to Neovim:
```

Comment:
I would write the enhancements like this:

- File format that

    - supports forward (ShaDa files from newer NeoVim can be used by older versions) and backward (ShaDa files from older versions can be used by newer ones without problems) compatibility, making ShaDa files future-proof to a great extent;
    - supports data structures which allow to store more kinds of data naturally;
    - is based on msgpack and explicitly standardized in documentation which allows creation of plugins/tools that perform arbitrary manipulations;
    - assumes no state stored between data pieces inside the file, which makes such tools simpler and allows such manipulations as “to concatenate two ShaDa files from different NeoVim instances simply use `cat 1.shada 2.shada > joined.shada`, NeoVim will handle this properly when reading”;
    - uses absolute paths.

- Embedded timestamps that allow multiple NeoVim instances to correctly merge their data when writing or reading ShaDa files.
- Forward compatibility includes NeoVim core code that preserves additional information found in ShaDa files generated by newer NeoVim versions in case they provide any.

About state: viminfo files *had* state:

1. Encoding (:h viminfo-c).
2. Lines with local marks looked like `> {file name}` (first line), `\t{mark}\tline\tcol` (subsequent lines). This is also why you should not talk about “efficient storage”. Viminfo was also very efficient, especially given that it needed no timestamps, it knew that it can end history string with NL (I still know it, but msgpack has no such a thing as “strings terminated by X”, it only use length-prefixed variants) and it used tab-separated fields where I have msgpack MAP.